### PR TITLE
InvalidLinkBear.py: Amend regex, ignore ``%s``,etc 

### PR DESCRIPTION
**Pull request by amrandahvur**
Modify regex to include links with `%`
in between and to ignore placeholders
like `%s`,`%c`,`%f`, etc. and add corresponding
tests to InvalidLinkBearTest.py

Closes https://github.com/coala/coala-bears/issues/758

### CODE REVIEW

**Comment by amrandahvur** [comment]
@sils I have incorporated the following changes:
1) Add ``http://www.google.com/hello%20world/200`` and ``http://httpbin.org/status%20is/200`` to the list of valid URLs
2) Removed ``http://www.google.com/hello%20world`` from the list of Invalid URLs since it has no reason to be invalid except for ``/200`` not being there.
3) Modified Regex so that tests pass

File: tests/general/InvalidLinkBearTest.py [hunk 1]
```diff
@@ -82,6 +82,8 @@ def test_run(self):
         valid_file = """
         http://httpbin.org/status/200
         http://httpbin.org/status/201
+        http://httpbin.org/status%20is/200
+        http://www.google.com/hello%20world/200
```
Comment (sils) [inline tests/general/InvalidLinkBearTest.py]: this line is redundant, also you should add a test for the invalid ones

Comment (amrandahvur) [inline tests/general/InvalidLinkBearTest.py]: Which line specifically ? In the review, you seem to have highlighted both the lines I have added i.e. ``http://httpbin.org/status%20is/200`` and ``http://www.google.com/hello%20world/200``.
Ok, adding a test for the invalid URLs as well :)

Comment (sils) [inline tests/general/InvalidLinkBearTest.py]: line86

Comment (sils) [inline tests/general/InvalidLinkBearTest.py]: the mocking doesn't care bout it being google or httpbin

**Comment by sils** [comment]
the commit messag doesn't describe your change

File: tests/general/InvalidLinkBearTest.py (hunk 1 above)
Comment (amrandahvur) [inline tests/general/InvalidLinkBearTest.py]: Oh, Okay :D Updating the PR now

**Comment by amrandahvur** [comment]
@sils Updated, Please Check ?

**Comment by sils** [comment]
please check your commit shortlog, "amend regex" doesn't really tell me what you did. What did you do?

File: bears/general/InvalidLinkBear.py [hunk 2]
```diff
@@ -44,7 +44,8 @@ def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
             r'((ftp|http)s?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'\\]+|\([^\s()\'\"`<>|\\]*\))%?[0-9]*(?:[^\s()%\'\"`<>|\\]+|'
+            r'\([^\s()\'\"`<>|\\]*\))*)(?<!\.)(?<!,)')
```
Comment (sils) [inline bears/general/InvalidLinkBear.py]: why are you almost doubling the size of the regex? So much complication?

**Comment by sils** [comment]
Hi,

please don't ping me personally for such things. Everyone can do them - and almost everyone pings me. If you don't address a person specifically you'll likely get help faster from other people!

(Don't worry, this is not a problem but keeping things general is usually a good policy :))

File: bears/general/InvalidLinkBear.py (hunk 2 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: I tried different things and only this seemed to work, Looks like I will have to look for another solution :/

Comment (sils) [inline bears/general/InvalidLinkBear.py]: the thing is that if something just seems to work in the given testcases than it'll likely be buggy unless you understand fully *why* it works

**Comment by amrandahvur** [comment]
Okay, Give me some time to figure out some optimal solution for regex :)

**Comment by sils** [comment]
sure

**Comment by sils** [comment]
sure!

**Comment by amrandahvur** [comment]
I need some more time, There are a lot of Deadlines I am currently having in my Institute. Please give me some days if possible :)

**Comment by Makman2** [comment]
@amrandahvur sure no problem :)

**Comment by amrandahvur** [comment]
Logistics:
1) Modified Regex to exclude links with placeholders like ``%s`` in URLs
The solution is a bit complex and messy as there are lot of combinations. Couldn't come up with a simpler one :(
2) Added an Invalid Test Case too
3) Changed Commit Shortlog to be as appropriate as possible (limit of 50 characters)

**Comment by Adrianzatreanu** [comment]
apparently we have merged something on this regex just now (removed the ftp, it doesnt affect you) but you need to rebase and fix the conflict :)

**Comment by amrandahvur** [comment]
@Adrianzatreanu Done :)

File: bears/general/InvalidLinkBear.py [hunk 3]
```diff
@@ -43,8 +43,9 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
```
Comment (Adrianzatreanu) [inline bears/general/InvalidLinkBear.py]: better shortlog: "Edit regex to ignore placeholders"

File: tests/general/InvalidLinkBearTest.py [hunk 4]
```diff
@@ -123,7 +124,7 @@ def test_run(self):
         http://httpbin.org/status/410
         http://httpbin.org/status/500
         http://httpbin.org/status/503
-        http://www.google.com/hello%20world"""
```
Comment (Adrianzatreanu) [inline tests/general/InvalidLinkBearTest.py]: is this considered to be a valid link actually? or why do you remove it?

File: bears/general/InvalidLinkBear.py (hunk 3 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: This cannot be done in 50 characters. (Because the filename occupies some space too) What should I do ? 
``InvalidLinkBear.py: Edit regex ignore placeholders``
Is this fine ?

Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: InvalidLinkBear: Allow placeholders in regex

File: bears/general/InvalidLinkBear.py [hunk 5]
```diff
@@ -43,8 +43,9 @@ def get_status_code(url, timeout):
...
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'\"`<>|\\]+|'
```
Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: Wow, this is complicated.
Can we please split this into sections and explain each section ?
```python
            r'(https?://'  # http or https
            r'[^.:%\s_/?#[\]@\\]+'  # Initial part of the domain
            r'\.'  # A required dot (.)
            r'(?:[^\s()%\'\"`<>|\\]+|\([^\s()%\'\"`<>|\\]*\))' # whatsit ?
            r'%?[0-9]*(?:[^\s()%\'\"`<>|\\]+|' # eh ?
            r'\([^\s()%\'\"`<>|\\]*\))*)(?<!\.)(?<!,)') # ooh
```

**Comment by jayvdb** [comment]
Maybe switch to regex whitespace mode so it can be presented logically.

**Comment by amrandahvur** [comment]
Updating the PR, give me some time :)

File: bears/general/InvalidLinkBear.py [hunk 6]
```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
...
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://'  # http:// or https://
+            r'[^.:%\s_/?#[\]@\\]+'  # Initial Part of Domain
+            r'\.'  # A required dot (.)
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Domain / Subdomain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))'
+            # Subdomain Name that is contained within ()
+            r'%?[0-9]*'  # A % sign (0 or 1 times) followed by any number
+                        # Thus %20 is allowed and %s isn't allowed
```
Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: Sorry for being pedantic - an extra space needed here

File: bears/general/InvalidLinkBear.py [hunk 7]
```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
...
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://'  # http:// or https://
+            r'[^.:%\s_/?#[\]@\\]+'  # Initial Part of Domain
+            r'\.'  # A required dot (.)
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Domain / Subdomain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))'
+            # Subdomain Name that is contained within ()
+            r'%?[0-9]*'  # A % sign (0 or 1 times) followed by any number
```
Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: So, if no letters are allowed, are we assuming a google search like "cpp %s error" is not a valid link that we will check ?
Example: https://encrypted.google.com/search?hl=en&q=cpp+%s+error

If this is expected, can this be added as a test case with a comment stating this ?

File: bears/general/InvalidLinkBear.py [hunk 8]
```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
...
+                        # Thus %20 is allowed and %s isn't allowed
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Same Regex for Domain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))*)(?<!\.)(?<!,)')  # Same Regex for
```
Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: Thanks a lot for the break up.

This now makes it clear that we're using `[^\s()%\'\"`<>|\\]` 4 times !
Can we store it in a variable like `domain_re = r'[^\s()%\'\"`<>|\\]'` and use it to avoid 4 changes if we want to change that regex ?

Note that this is not in the scope of this PR (as you solve the issue). So, preferably, just open a new github issue and mark it as newcomer so that a newcomer can solve it!

File: tests/general/InvalidLinkBearTest.py [hunk 9]
```diff
@@ -82,6 +82,7 @@ def test_run(self):
         valid_file = """
         http://httpbin.org/status/200
         http://httpbin.org/status/201
+        http://httpbin.org/status%20is/200
```
Comment (AbdealiJK) [inline tests/general/InvalidLinkBearTest.py]: Please add test cases along with the commit that needs these test cases :)
Squish both the commits together ?

File: bears/general/InvalidLinkBear.py (hunk 8 above)
Comment (Makman2) [inline bears/general/InvalidLinkBear.py]: if he already did this he can merge this in a row with this PR imo (in a separate commit obviously :smile:)

File: bears/general/InvalidLinkBear.py (hunk 7 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Hey, So, I guess I shall ask the same, Would you want me to consider ``https://encrypted.google.com/search?hl=en&q=cpp+%s+error`` as valid or invalid ? What should it be ideally ?

File: bears/general/InvalidLinkBear.py (hunk 6 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Extra space where specifically ? Line Number ? I guess its 54 right ? For the alignment of ``#`` ?

File: bears/general/InvalidLinkBear.py (hunk 7 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Because according to what you guys want, I shall try to modify the regex although its getting complicated already.

Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: Hm. I think @sils needs to comment here

File: bears/general/InvalidLinkBear.py (hunk 6 above)
Comment (AbdealiJK) [inline bears/general/InvalidLinkBear.py]: Yes, for the alignment of the # - sorry about the ambiguity :D

File: bears/general/InvalidLinkBear.py (hunk 7 above)
Comment (sils) [inline bears/general/InvalidLinkBear.py]: `%` is `%25` so your link should be https://encrypted.google.com/search?hl=en&q=cpp+%25s+error if you want it to be recognized

Comment (sils) [inline bears/general/InvalidLinkBear.py]: because if it's a %s the user might want to actually replace it there as well and do a format on that

Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Cool, Thank you, I will be updating the PR soon :)

**Comment by amrandahvur** [comment]
Okay, So, I have figured out a regex that works for all possible combinations.
Here is the link https://regex101.com/r/MrDBzP/8
But when I run the tests, they raise an error.


Why is this an error ? ``AssertionError: [ ] == [ ]`` ?
After all, two empty lists are always the same right ?

**Comment by sils** [comment]
it's an assert *not* equal

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

**Comment by amrandahvur** [comment]
Yeah, I figured that out, thus deleted my previous comment. This is taking time although, Its really complicated :(

**Comment by Makman2** [comment]
don't give up, if something's unclear just ping us :) :+1:

**Comment by amrandahvur** [comment]
@Makman2 Thanks for sticking with me :)
Okay, So here is the regex ``https://regex101.com/r/MrDBzP/13`` and it also contains all the testcases for the ``test_run`` function and it works well :) as the tests pass.
But the problem occurs with ``test_link_ignore_regex`` function in InvalidLinkBearTest.py
Here is the error:

How can I possibly solve this now ?

**Comment by Makman2** [comment]
honestly I don't understand the meaning of this particular test, the mocks made there are too weird for me :3
@sils why the heck some of those links shall be ignored?

**Comment by sils** [comment]
@Makman2 usually they're on the ignore list because they have placeholders or it's likely that something's replaced and the URL isn't valid on it's own

**Comment by amrandahvur** [comment]
So what shall be done finally (after using non-capturing groups) ? The issue with the ``test_link_ignore_regex`` function has to be resolved ultimately.
Also, The URLs listed in this function namely:
``
http://sub.example.com
http://sub.example.com/something
``
AND
``
http://www.notexample.com
http://exampe.com
http://example.co.in
``
Do they need to pass the regex (a successful match)?
Also, should I add a status code at the end of each URL?
Example: ``http://sub.example.com/202``

**Comment by Makman2** [comment]
@sils have you looked at the test links? they are completely valid imo^^

**Comment by Makman2** [comment]
@sils ping

**Comment by amrandahvur** [comment]
The Tests pass on my local machine, So they should pass on CircleCI too right ??

**Comment by Makman2** [comment]
@amrandahvur your coverage is not 100%, maybe this is the fault?
https://github.com/coala/coala-bears/pull/932/commits/6db43de3a5ea7e42f9c2ce1d6817e09917b59cb9#diff-c4521389f8a26d35c040551df14b3557L129

**Comment by Makman2** [comment]
@amrandahvur btw you can see coverage report locally when appending `--cov` to your test execution command :)

**Comment by amrandahvur** [comment]
@Makman2 Well, Is this because I added the status code ``500`` to ``http://coalaisthebest.com/`` ?
Although the regex makes links with placeholders (present anywhere in the URL) invalid, but it also demands that the URL ends with a status code. Thus I had to add it :/
How do I solve this issue now pheww :/

**Comment by Makman2** [comment]
@amrandahvur just remove the `500` again and run tests locally with coverage report and see whether coverage is complete. I'm not sure if this is the problem, was just a first guess^^

**Comment by amrandahvur** [comment]
@Makman2 So here are the results 
1) With ``500``, Coverage is 97% and tests pass
2) Without ``500``, Coverage is 95% and tests fail

**Comment by Makman2** [comment]
okay then this is not the cause. Then try to add another test case which covers the missing lines :+1:

**Comment by sils** [comment]
btw. please consult me before merging this PR, I don't have time to watch
over this the whole time but I want to see to it that it's done properly

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

**Comment by amrandahvur** [comment]
@Makman2 Fixed all errors :)
The coverage is also 100% now :)

**Comment by jayvdb** [comment]
Also it would be good if https://github.com/coala/coala-bears/issues/1011 was fixed first.

**Comment by Makman2** [comment]
@jayvdb @amrandahvur you don't have to fix #1011, this is just overhead work. Tests do work, beautification can be done later^^

**Comment by jayvdb** [comment]
@Makman2 , it is not overhead. This patch adds comments to the regex. Those comments were wrong, and the existing regex wasnt understood. Rather than drown *this* PR with review comments about what the regex is currently doing, that processis being done on the other PR, and it will mean this PR is easier to review.

**Comment by Makman2** [comment]
@jayvdb #1011 is about improving the mocking stuff of the current `InvalidLinkBearTest`. I think you are talking about something else :)

**Comment by amrandahvur** [comment]
@jayvdb Since https://github.com/coala/coala-bears/issues/1018 has been closed now, Could you please remove ``status/blocked`` tag ? I would like to resume my work on this :D

**Comment by sils** [comment]
sure

**Comment by amrandahvur** [comment]
All the tests pass with 100% coverage, What can possibly be the error for failure of CircleCi ? The log shows it has to do with something about ``pyflakes`` and stuff

**Comment by aptrishu** [comment]
@amrandahvur try `pip install "pyflakes!=1.2.0,!=1.2.1,!=1.2.2,<1.3.0,>=0.8.1"`

**Comment by aptrishu** [comment]
rebase and push again..

**Comment by amrandahvur** [comment]
@aptrishu Done, Still the same :/

**Comment by aptrishu** [comment]
still the pyflakes version you have is not up to what is required.. search for the version in requirements.txt.

**Comment by jayvdb** [comment]
(rebuilding without cache)

**Comment by amrandahvur** [comment]
I can't understand :/
``requirements.txt`` tells me to install ``pyflakes~=1.3``
while the failed tests of CircleCi tells me to install using ``pip install "pyflakes!=1.2.0,!=1.2.1,!=1.2.2,<1.3.0,>=0.8.1"``
What should be done ?
Is there an escape to this ? :(

File: bears/general/InvalidLinkBear.py [hunk 10]
```diff
@@ -57,24 +57,64 @@ def find_links_in_file(file, timeout, link_ignore_regex):
...
             \.                          # A required dot `.`
             (
-                (?:[^\s()%\'"`<>|\\]+)  # Path name
+                (?:[^\s()%\'\"`<>|\\]+) # Path name
                                         # This part does not allow
                                         # any parenthesis: balanced or
                                         # unbalanced.
             |                           # OR
-                \([^\s()%\'"`<>|\\]*\)  # Path name contained within ()
+                \([^\s()%\'\"`<>|\\]*\) # Path name contained within ()
```
Comment (jayvdb) [inline bears/general/InvalidLinkBear.py]: the commit summary doesnt explain this change.
I believe it is a separate problem?  If so, it should be a separate PR, with a separate commit message.

**Comment by jayvdb** [comment]
Re pyflakes, somehow we have 'flake8' in the CircleCI cache, and it is causing the pyflakes problems.  We dont need flake8, but it is being installed anyway ;-(

A fresh build without cache and it works : https://circleci.com/gh/coala/coala-bears/4176

File: bears/general/InvalidLinkBear.py (hunk 10 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Well, This isn't a major issue. We are just escaping ``"`` with a backslash (``\``). Is it still necessary for a separate PR ?

**Comment by sils** [comment]
Can be in the same pr as long as it's an own commit

**Comment by amrandahvur** [comment]
@sils @jayvdb Please have a look

File: bears/general/InvalidLinkBear.py (hunk 10 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: It's just that when I tested the regex on ``regex101.com``, It shows an error unless I escape it, Should I still remove the commit ?

Comment (Makman2) [inline bears/general/InvalidLinkBear.py]:
```
Python 3.5.2 (default, Sep 14 2016, 11:28:32) 
[GCC 6.2.1 20160901 (Red Hat 6.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> re.compile('"')
re.compile('"')
>>> 
```
no errors. I think this is because regex101 assumes that you wrap your regex string inside double quotes.

Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Cool, Removing it :D

File: bears/general/InvalidLinkBear.py [hunk 11]
```diff
@@ -68,13 +68,53 @@ def find_links_in_file(file, timeout, link_ignore_regex):
...
                                         # prevents matching of
                                         # Unbalanced parenthesis
+            (
+                (%?[0-9]+)              # A ``%`` sign (0 or 1 times)
+                                        # followed by any number
+                                        # Thus ``%25`` is allowed
+                                        # and ``%s`` isn't allowed
+                                        # This also puts a restriction
+                                        # that the URLs have to end with
+                                        # a number (HTTP Status Code)
```
Comment (Mixih) [inline bears/general/InvalidLinkBear.py]: These examples could be removed to thin down this  comment string leaving
````
# A ``%`` sign (0 or 1 times)
# followed by any number
````

File: bears/general/InvalidLinkBear.py [hunk 12]
```diff
@@ -68,13 +68,53 @@ def find_links_in_file(file, timeout, link_ignore_regex):
...
+            )                           # This is the first part of the regex
+                                        # that covers URLs with or without
+                                        # placeholders but with necessary
+                                        # HTTP Status Codes at the end
+            |                           # OR (Second part of regex begins here)
+                                        # This covers URLs without any
+                                        # numbers at the end
+                                        # This is essential for URLs
+                                        # with placeholders but
+                                        # without HTTP Status Codes
```
Comment (Mixih) [inline bears/general/InvalidLinkBear.py]: see if you can thin down line 90-99 if possible

File: bears/general/InvalidLinkBear.py (hunk 11 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: Ok :+1: Removing it :smile:

File: bears/general/InvalidLinkBear.py (hunk 12 above)
Comment (amrandahvur) [inline bears/general/InvalidLinkBear.py]: But I should let the meaning be intact right ? I will try to re-frame these phrases :+1:

File: bears/general/InvalidLinkBear.py (hunk 11 above)
Comment (Mixih) [inline bears/general/InvalidLinkBear.py]: Wait you might want to keep 
````
#This also puts a restriction
# that the URLs have to end with
# a number (HTTP Status Code)
````
in some shorter 2 line form...

**Comment by sils** [comment]
@amrandahvur this seems to be good work but TBH the regex seems to grow way too complex, I'm having problems understanding it and that (usually) means others will as well. That will lead to people changing it and breaking it and also makes it hard to determins if it's already broken.

So I really think we gotta make this better maintainable. Maybe we can put parts of the regex in variables to reduce the duplications and give them names==meanings?

**Comment by amrandahvur** [comment]
@sils Yes, there are many parts of the regex that are repeated. We can store them in variables and reuse them which will make it more maintainable and also reduce duplications. The question is, Should I do it or perhaps I could create a new issue out of it ?

File: bears/general/InvalidLinkBear.py [hunk 13]
```diff
@@ -68,13 +68,49 @@ def find_links_in_file(file, timeout, link_ignore_regex):
...
+            (
+                (?:[^\s()%\'"`<>|\\]+)  # Same Regex for Path Name
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Same Regex for Path name
+                                        # contained within ()
+            )
+            ))                          # Thus, the whole part above
+                                        # prevents matching of
+                                        # Unbalanced parenthesis
+            (?!(.))(?<!\.)(?<!,))"""    # Negative lookahead to match URLs not
```
Comment (jayvdb) [inline bears/general/InvalidLinkBear.py]: please move the closing `"""` to the same position as the previous regex, before `, re.VERBOSE)`

File: bears/general/InvalidLinkBear.py [hunk 14]
```diff
@@ -68,13 +68,49 @@ def find_links_in_file(file, timeout, link_ignore_regex):
...
-                                        # Thus, the whole part above
+            )                           # Thus, the whole part above
                                         # prevents matching of
                                         # Unbalanced parenthesis
+            (
+                (%?[0-9]+)              # A ``%`` sign (0 or 1 times)
+                                        # followed by any number
+                                        # Restricts URLs to end with
+                                        # a number (HTTP Status Code)
+                [^\s()%\'"`<>|\\]+      # Same Regex for Path Name
```
Comment (jayvdb) [inline bears/general/InvalidLinkBear.py]: Regex and Path Name should be all lower case (here and elsewhere, only use capitals for start of sentence , or proper nouns , or abbreviations).

**Comment by amrandahvur** [comment]
@jayvdb Updated :)

**Comment by Mixih** [comment]
I'd suggest you make variables now, at least for the parts you annotated.

**Comment by Nosferatul** [comment]
closing due to high inactivity
